### ISSUE TITLE
GL/Vulkan: Grand Theft Auto V graphical issues

### ISSUE THREAD

**Issue by MarioSonic2987** [RPCS3/rpcs3#3959]
Graphical issues:

- [x] **[Distorted UI elements (HUD, loading screens...)](https://user-images.githubusercontent.com/20027681/39252326-b5971df4-48a5-11e8-965a-89697fa09d50.png)** - Fixed in #10453

- [x] **[Black textures](https://user-images.githubusercontent.com/20027681/39252440-ef430432-48a5-11e8-9a5a-ea43bcebeb9f.png)** - Fixed in #6415 

- [x] **Dark shadows** - Fixed in #8186

- [x] **Broken ambient occlusion** - Fixed in #8713

- [x] **Trevor's outfit changes color (red or black) and other characters during prologue's mission** - Fixed.

- [x] **Broken foliage in some locations** - Fixed in #6228 

- [x] **No color in skybox** - Fixed in #8243

- [ ] **Broken reflections** (Only present in Vulkan.)

- [x] **Weird tracing when rotating the camera** - Fixed in #6415 

[Footage (June 25, 2019)](https://www.youtube.com/watch?v=R4OYd4dqoyI)

[Log OpenGL](https://github.com/RPCS3/rpcs3/files/1947471/Log.GTA.V.25-4-18.GL.zip)
[Shaderlog OpenGL](https://github.com/RPCS3/rpcs3/files/1947476/Shaderlog.GTA.V.25-4-18.GL.zip)
[RenderDoc OpenGL (Loading screen and ingame)](https://drive.google.com/open?id=1BNr0fERWaboxjtFTqEmB7zAbkw5UEl1r)

<details>
 <summary>Original comment</summary>
With https://github.com/RPCS3/rpcs3/commit/a08e79f4267b5e9fab82949a5f011eb40aca76b0, GTA V renders its UI like this (look at the Rockstar's logo):

![image](https://user-images.githubusercontent.com/20027681/34226708-d1516c0e-e5ca-11e7-8f34-9cdd7b6dd24e.png)

But with RenderDoc, I was able to modify the shaders. I went to the vertex one to investigate what was causing that bug. And I found these two lines of code:

![rd1](https://user-images.githubusercontent.com/20027681/34226797-23e35b4e-e5cb-11e7-8444-8b7f7ced9913.png)
By changing bits 216 and 217, it fixed Rockstar's logo's orientation somehow, but the rest of UI is moved and rotated.
![rd2](https://user-images.githubusercontent.com/20027681/34226802-29bafb62-e5cb-11e7-8659-1e7e55bb0c12.png)

In conclussion, this bug is caused, likely, by a vertex shader instruction badly implemented.

[Log OpenGL](https://github.com/RPCS3/rpcs3/files/1576916/Log.GTA.V.20-12-17.GL.gz)
[Shaderlog OpenGL](https://drive.google.com/open?id=1oLvU0iv8DFNTtUcKdsK7wrvoxOaOMt7T) - It's important the latest generated vertex shader.
[RenderDoc OpenGL](https://drive.google.com/open?id=15jQKO-h5_8cyt_mcT5ym5D4L_mJWwUcm)
</details>

**Comment by kd-11** [comment]
Actually that change just changes referenced data that is uploaded from the cpu. There is no ARL preceding it meaning the values are intentionally that way (also confirmed in that swapping them rotated everything). Might be a vertex register misalignment or something broken in cell side calculations.

**Comment by MarioSonic2987** [comment]
With https://github.com/RPCS3/rpcs3/commit/345f92ab0a57a4f302f6fac651372bcf42005080:

This screenshot is taken with PPU interpreter fast and SPU interpreter precise, and there's no graphic difference.


[RSX Capture Loading screen (Vulkan)](https://github.com/RPCS3/rpcs3/files/2149623/capture.GTA.V.LS.29-6-18.zip)


![gta v-ppu spu prec](https://user-images.githubusercontent.com/20027681/42095762-d02bb198-7bb3-11e8-942d-2c446314deac.png)

**Comment by MarioSonic2987** [comment]
Ace Combat Infinity, like GTA V, has the same distorted UI bug. With these settings and with/without Force CPU blit, it doesn't improve.

Screenshot of the bug:
![](https://cdn.discordapp.com/attachments/361063300487970818/501063352387239936/unknown.png)


[RSX Capture (ACI and GTA V)](https://github.com/RPCS3/rpcs3/files/2476599/RSX.Capture.14-10-18.zip)

**Comment by MarioSonic2987** [comment]
I've attached the latest capture of the distorted UI glitch.
![image](https://user-images.githubusercontent.com/20027681/57539119-121a5300-734a-11e9-9049-cd8c8bbaf4a1.png)



[RSX Capture (Pause menu)](https://github.com/RPCS3/rpcs3/files/3167178/NPEB01283_20190510173442_capture.zip)

**Comment by MarioSonic2987** [comment]
This game needs Write Color Buffers to fix lighting outside.

<details>
 <summary>Screenshot comparison</summary>

**Without WCB**:
![image](https://user-images.githubusercontent.com/20027681/62364410-acd99680-b521-11e9-93f8-62e93ea8aa7f.png)
**With WCB**:
![image](https://user-images.githubusercontent.com/20027681/62364416-af3bf080-b521-11e9-96da-6cc7087e6db2.png)

</details>

**Comment by MarioSonic2987** [comment]
I've attached a RDC in Los Santos to show the graphical issues
![image](https://user-images.githubusercontent.com/20027681/63328582-affdc080-c330-11e9-92de-f24c534c5df4.png)

[RenderDoc VK](https://drive.google.com/open?id=1n5gViyl7K5ol209_MWlIGVZtpMyw3d2u)

**Comment by MarioSonic2987** [comment]
I've attached a OpenGL renderdoc trace for the dark shadows. Tested with 0.0.7-8700-32a4c224.

[RenderDoc OpenGL](https://drive.google.com/open?id=1KOIIfpzaKQYBWiwmTzy3-QD1X-1FNLef)

**Comment by MarioSonic2987** [comment]
Raindrops are rendered by using WCB+WDB+RCB+RDB.

<details>
<summary>Missing raindrops</summary>
![image](https://user-images.githubusercontent.com/20027681/71645657-b8510980-2cdb-11ea-9a7b-ab8ac005147a.png)
![image](https://user-images.githubusercontent.com/20027681/71645658-bb4bfa00-2cdb-11ea-86f6-b43675ce0548.png)
</details>

Models no longer turns black in certain camera angles. But the red textures are still there:
![image](https://user-images.githubusercontent.com/20027681/71645671-0403b300-2cdc-11ea-8ac0-1b9f82eb4d77.png)
![image](https://user-images.githubusercontent.com/20027681/71645677-2b5a8000-2cdc-11ea-9c40-98278a533e72.png)

**Comment by The-SamminAter** [comment]
I'm getting severely distorted/laggy audio, and for some reason characters are red.
![FPS_ 16 54 _ Vulkan _ 0 0 8-9349 _ Grand Theft Auto V  BLES01807  2020-01-15 5_10_52 PM](https://user-images.githubusercontent.com/48739810/72484722-11c83680-37ba-11ea-9e70-810a2841b460.png)

**Comment by MarioSonic2987** [comment]
Reflections are broken, like in these puddles.
![image](https://user-images.githubusercontent.com/20027681/81585692-cebb5000-93b4-11ea-85fa-3bf9e5e2eeef.png)
![image](https://user-images.githubusercontent.com/20027681/81585956-2954ac00-93b5-11ea-8aa8-ef95cad4ba7f.png)

**Comment by MarioSonic2987** [comment]
Sea isn't vertex-exploded if you disable WCB, as well the weird garbage glitch:
![image](https://user-images.githubusercontent.com/20027681/82230621-4268da00-992c-11ea-8d73-55472dd19d6a.png)
![image](https://user-images.githubusercontent.com/20027681/82230661-54e31380-992c-11ea-8ba8-c5bbaff833e8.png)

With WCB enabled:
![image](https://user-images.githubusercontent.com/20027681/82231082-eeaac080-992c-11ea-8dd4-941b5bda7645.png)
![image](https://user-images.githubusercontent.com/20027681/82231061-e8b4df80-992c-11ea-9a16-9696cdb173c8.png)

**Comment by xddxd** [comment]
Ambient occlusion seems to be broken as well.
<details>
  <summary>Prologue shots</summary>
  
RPCS3
![Screenshot_4](https://user-images.githubusercontent.com/1712979/84901601-6434b880-b0b4-11ea-9d8d-488661459c6e.png)
RealHW shot from Youtube
![Screenshot_6](https://user-images.githubusercontent.com/1712979/84901626-6d258a00-b0b4-11ea-9fbf-d95044623f69.png)
PC
![sspcprologue](https://user-images.githubusercontent.com/1712979/84901652-73b40180-b0b4-11ea-8a7a-e2f79d81b31a.png)
  
</details>
<details>
  <summary>Franklin's house</summary>
  
RPCS3
![Screenshot_3](https://user-images.githubusercontent.com/1712979/84901697-83cbe100-b0b4-11ea-8578-0cabb07b90eb.png)
PC
![sspc](https://user-images.githubusercontent.com/1712979/84901722-8cbcb280-b0b4-11ea-94ce-8681060b0060.png)
  
</details>
Log - [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/4792648/RPCS3.log.gz)
PC and RPCS3 renderdocs https://www.dropbox.com/s/7jj8lff29mc5yne/gtavrdc.7z?dl=0

**Comment by kd-11** [comment]
For post-processing missing bugs, please make sure to test with all the buffer options enabled in case the game is using SPUs for some effects.

**Comment by kd-11** [comment]
And also accurate xfloat just in case. There's a very large chance this effect is not done GPU-side given how weak the shader pipes are on RSX.

**Comment by xddxd** [comment]
In the image WCB+WDB+RCB+RDB and accurate xfloat are enabled, no change.
![image](https://user-images.githubusercontent.com/1712979/84906805-15d6e800-b0bb-11ea-8e21-a6fd9dd9d22c.png)
In addition to the settings mentioned above, I tried SRM and CPU blit as well, no change.

**Comment by MarioSonic2987** [comment]
Broken ambient occlusion is now fixed.

![image](https://user-images.githubusercontent.com/20027681/92310194-30c74300-efac-11ea-99b6-1f3551a11246.png)
![image](https://user-images.githubusercontent.com/20027681/92310195-3de43200-efac-11ea-80ea-930f276b945d.png)

**Comment by MarioSonic2987** [comment]
I attached two RSX Captures for the broken UI bug.

[RSX Captures (Pause menu and brightness calibration screen)](https://drive.google.com/file/d/1ZQW0MSbp_UbOYvuOdIwYvBvTIMb7a-cu/view?usp=sharing)

**Comment by Xcedf** [comment]
Dudes don't turn red anymore during intro mission
https://youtu.be/LE8C-12tIKw
unfortunately game froze during capture, but still

**Comment by MarioSonic2987** [comment]
@Xcedf I can confirm that the red texture bug is fixed, even if you get injured.
![image](https://user-images.githubusercontent.com/20027681/93688234-97785080-fac4-11ea-8825-8acea12473d2.png)
![image](https://user-images.githubusercontent.com/20027681/93688320-3c932900-fac5-11ea-887a-278edea08b46.png)

But it still happens with OpenGL.
![image](https://user-images.githubusercontent.com/20027681/93688723-289cf680-fac8-11ea-8df9-e32113df2e77.png)

Cellphone screen has now correct colors. Previously, colors were inverted.
![image](https://user-images.githubusercontent.com/20027681/93688262-ddcdaf80-fac4-11ea-942b-b7d3371d7d29.png)
![image](https://user-images.githubusercontent.com/20027681/93688218-71eb4700-fac4-11ea-9cf4-9859d56a08fe.png)

Although car reflections are missing in OpenGL. Vulkan shows them correctly.
![image](https://user-images.githubusercontent.com/20027681/93688764-95b08c00-fac8-11ea-8ecc-938f4eb65ba5.png)
![image](https://user-images.githubusercontent.com/20027681/93688839-43239f80-fac9-11ea-9dbe-4650adc5a200.png)

**Comment by Xcedf** [comment]
@MarioSonic2987 can't get to Franklin mission in the car shop cause of freezing, maybe Setarian strange monkey face bug is also fixed. Besides do you know maybe game need some special settings to maintain stability?

**Comment by MarioSonic2987** [comment]
@Xcedf You need to change Driver Wake-Up Delay to, at least, 400 μs.

The "monkey face" bug is still there, but it happens when a character is too close to the camera. It also happens in the prologue in the same manner (see Michael's jacket in my previous comment).
![image](https://user-images.githubusercontent.com/20027681/93715019-e7a7ef00-fb66-11ea-8b67-82815cda63b3.png)
![image](https://user-images.githubusercontent.com/20027681/93715034-fd1d1900-fb66-11ea-96df-33b728acb1ae.png)

**Comment by Xcedf** [comment]
@MarioSonic2987 Freezes for me even on 5000μs.

**Comment by Darkhost1999** [comment]
The audio if it helps is an Echo of the explosion or "fuc*!!!" That is shouted. Any loud noise is amplified in an echo while distorted.

**Comment by Xcedf** [comment]
Water in the ocean seems to be no longer vertex exploding
recent master build, wawes are fine
https://youtu.be/yW397eIr-_U
for example some older build, feels like it's boiling
https://youtu.be/xN0pUXDlx9o
Reflections are still mess, but water in the pool no longer radioactive on Vulkan as well
![image](https://user-images.githubusercontent.com/7060586/127747068-bc1ff03f-cca3-45d4-a459-1484973f8869.png)

**Comment by MarioSonic2987** [comment]
I can confirm water doesn't freak out in certain camera angles:
![image](https://user-images.githubusercontent.com/20027681/127747632-d07e74b3-9d9a-490b-82f5-feefd3d9d9da.png)

But reflections are still borked.
![image](https://user-images.githubusercontent.com/20027681/127747676-c05b2421-e758-4876-906e-b6be3ff86fd1.png)

However, when you dive into the water, it looks fine.
![image](https://user-images.githubusercontent.com/20027681/127747693-df92610c-8cad-4380-b3cd-8934b7d24725.png)

**Comment by MarioSonic2987** [comment]
Broken reflections in vehicles and puddles in OpenGL is a regression caused by PR #8710.

**Comment by Xcedf** [comment]
@MarioSonic2987 you're saying that rainbow artifacts is an old regression, so it worked on Vullkan at some point?

**Comment by MarioSonic2987** [comment]
Rainbow blocky artifacts regression present in Vulkan are caused by PR #6359. Commit 91d711087f0d63e01c8e459815a6f7b37812f9a0.

**Comment by Darkhost1999** [comment]
Loud noises in GTA V persist with CubeB PR (recently merged)

**Comment by MarioSonic2987** [comment]
Characters no longer turn black in cutscenes:
![image](https://user-images.githubusercontent.com/20027681/148680394-df74a15e-ef7f-4881-b89c-7f7273afaeef.png)

**Comment by Darkhost1999** [comment]
So is the only thing remaining on this ticket for issues tracked here are missing reflections? If any other issues create a separate ticket?

**Comment by MarioSonic2987** [comment]
Not a graphical fix, but loud noises in most situations, like this cutscene, are no longer present.
![image](https://user-images.githubusercontent.com/20027681/168077905-92b82a51-dcfe-479e-83ce-77bc3a985a0f.png)

**Comment by Darkhost1999** [comment]
> Not a graphical fix, but loud noises in most situations are no longer present.

Indeed loud noises are still present at randomly occurring times but nowhere near the amount of frequency as before. Very rare for pops and bangs now.

**Comment by Xcedf** [comment]
@Darkhost1999 remove PPU Cache and try again, it was LLVM fix

**Comment by Darkhost1999** [comment]
> @Darkhost1999 remove PPU Cache and try again, it was LLVM fix

I have removed all of my caches and compiled them all again.
The loud pops and bangs are rare not gone.

**Comment by Darkhost1999** [comment]
In the end wrong issue ticket anyways.

**Comment by Xcedf** [comment]
Did you remember me asking to change or drop your point? I can't. I'm just asking to test this certain place in the game that's it, to make sure that at least this is working.

**Comment by Darkhost1999** [comment]
In the event it has not been said water is a regression.

Build with regression https://github.com/RPCS3/rpcs3/pull/6359

![Screenshot_20220803-002533.png](https://user-images.githubusercontent.com/60384196/182530670-97d2093f-6653-4240-9cfb-fcfff4ad054e.png)

Zerox used illusions build as a base. Then just reverted things from there.
https://discord.com/channels/272035812277878785/272035812277878785/934899723461066772
Although I still see no source code or anything to show what changes were made if you find this useful. Maybe the fix can be added to master soon

**Comment by Lucas7yoshi** [comment]
A regression occured, the game completely locks up at the "Get to the car" stage of prologue, FPS not changing

This has occured every time for 5 times.

0.0.24-14133
GTA 5 is updated with all updates to 1.27


Using all compatibility options described here https://wiki.rpcs3.net/index.php?title=Grand_Theft_Auto_V

Footage/log is from fresh cache
Video: https://www.youtube.com/watch?v=J6wHd1I3ccI
Log file: https://sx.l7y.ca/gta5_2022_09_08_lucas7yoshi_RPCS3_fH8nLmQz8.log.gz

**Comment by Darkhost1999** [comment]
>the game completely locks up at the "Get to the car" stage of prologue
Debugger wouldn't stop mentioning getllar so I enabled accurate RSX reservation access and it worked. I tested as far back as https://github.com/RPCS3/rpcs3/pull/12234 and gave up.
Just enable accurate RSX reservation access as soon as you're Franklin you don't "need" the setting anymore. But it also prevents freezing when swimming as anyone later in game.

**Comment by MarioSonic2987** [comment]
Tested on 0.0.24-14195.
<details>
 <summary>OpenGL issues</summary>

It has broken ambient occlusion, missing car reflections and minor vertex explosions in the water:
![image](https://user-images.githubusercontent.com/20027681/192593479-2600bf39-44a8-4e88-b477-36fd4c49b0db.png)
![image](https://user-images.githubusercontent.com/20027681/192600708-60d613c0-41d8-4bb7-8148-e0bdf4323a41.png)

Character's portrait in the pause menu is rendered black (correct in Vulkan):
![image](https://user-images.githubusercontent.com/20027681/192599725-bcfd698f-4ece-4208-a1c9-a85365239fbc.png)
![image](https://user-images.githubusercontent.com/20027681/192600302-b24ac4f8-f72b-491b-b8f1-1b3a2fa7d5e4.png)


Red models in certain conditions (when someone gets injured, for example) are no longer present:
![image](https://user-images.githubusercontent.com/20027681/192595838-b99a225b-edea-4d61-82e8-958b32c687df.png)
</details>


Vulkan-exclusive glitches (rainbow garbage and broken reflections) are still present.

**Comment by MarioSonic2987** [comment]
PR #12762 breaks graphics in Vulkan. OpenGL works fine.
![image](https://user-images.githubusercontent.com/20027681/194538811-8b8902d1-7922-428a-944e-7bae483c118b.png)

**Comment by MSuih** [comment]
Vulkan regression might be fixed by #12768

**Comment by MarioSonic2987** [comment]
That PR fixes this regression.
![image](https://user-images.githubusercontent.com/20027681/194553245-1d879789-fd01-4aa2-9068-1a3199c9cead.png)

**Comment by kd-11** [comment]
Check if water is fixed by https://github.com/RPCS3/rpcs3/pull/13070

**Comment by Ordinary205** [comment]
Intermittent vertex explosions on vehicles still occurs even on the latest master.

https://github.com/RPCS3/rpcs3/assets/125112932/1db0b39f-4e36-4c3d-9554-b46b966a4bf0

**Comment by Lethality5001** [comment]
Will water be fixed soon? Or car flickering when damaged enough

**Comment by kd-11** [comment]
Has anything changed here? Time for a retest.

**Comment by Lethality5001** [comment]
> Has anything changed here? Time for a retest.

I will test the current things again in a few hours and let u know

**Comment by MarioSonic2987** [comment]
Water still has vertex explosions:
![Image](https://github.com/user-attachments/assets/a837387b-50ce-4ecc-ab0e-64c0d565876d)

Distant reflections are still broken:
![Image](https://github.com/user-attachments/assets/ee186eb5-8cc7-4acb-8164-7642abc349b2)

Flickering rainbow artficats are still there.

[Log](https://github.com/user-attachments/files/18632716/RPCS3.log.gz)

[RSX Capture 1](https://drive.google.com/file/d/1g8qPUdE7lVOqBBj71razxVNJxA_Hm1Mi/view?usp=sharing)
[RSX Capture 2](https://drive.google.com/file/d/1S9BMbbDJSGh8PO-XM6TGfbvm3MGYi3WY/view?usp=sharing)

**Comment by Thunderstorm64** [comment]
> Intermittent vertex explosions on vehicles still occurs even on the latest master.
> Desktop.2023.08.19.-.19.54.06.02_Trim.mp4
> 
> [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12386281/RPCS3.log.gz)

Hmm the vertex explosions while driving only seem to happen if you damage the car but only while you're in it. You can damage it outside and it will be fine but if driving they start, and the more damage the worse they get

**Comment by rxhfcy** [comment]
> Has anything changed here? Time for a retest.

I upgraded my CPU (everything else stayed the same, same GPU, same RPCS3 version) from i5-12400F ("C Tier") to a slightly overclocked i7-13700K ("S Tier") and the vertex explosions while driving were FIXED completely, used to be glitches and explosions everywhere all the time.

So could that one be at least partly "just" a performance problem?

**Comment by kd-11** [comment]
Faster CPU is probably fixing the problem by "winning" some race conditions. That makes this a synchronization problem, meaning settings like "accurate RSX reservations" or those other SPU accuracy settings in advanced tab might have an impact.

**Comment by Thunderstorm64** [comment]
> > Has anything changed here? Time for a retest.
> 
> I upgraded my CPU (everything else stayed the same, same GPU, same RPCS3 version) from i5-12400F ("C Tier") to a slightly overclocked i7-13700K ("S Tier") and the vertex explosions while driving were FIXED completely, used to be glitches and explosions everywhere all the time.
> 
> So could that one be at least partly "just" a performance problem?

I noticed that they dont occur immediately upon loading in the game, but can take 10-20 (rough estimation) mins to start showing up

**Comment by CodeSavei** [comment]
Is the vertex explosion fix a big hurdle? I think it's the only bug that prevents it from being enjoyable. I love this version of the game.

**Comment by tarasdeinum** [comment]
yo i have the problem that how more the car damages how more it flickers how to fix

**Comment by matheuswillder** [comment]
> yo i have the problem that how more the car damages how more it flickers how to fix

Currently there is no way to fix this, this happens due to the current state of the emulation. That's precisely why this issue exists, to track emulation bugs.

**Comment by badcops2017** [comment]
Hoping the vertex explosions can be fixed in the future

**Comment by badcops2017** [comment]
Also the monkey face bug still occurs for me, i dunno if its some sort of setting i needa enable but ive tried almost everything to fix it and nothing happens

**Comment by digant73** [comment]
please re-test on latest build. Hopefully some textures and reflections issues have been fixed by recent build 0.0.39-18790

**Comment by MarioSonic2987** [comment]
Reflections are still broken.

<details>
 <summary>Images</summary>

<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/f37866d4-d44f-49dc-b296-f51249ca1426" />

<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/294419fa-caca-4275-92ce-8822eecbacf9" />

</details>

**Comment by Xcedf** [comment]
Main character no longer has vertex explosions  with water reflections, perhaps car reflections also better, and dark monkey face is quite recent regression, will track it down.

**Comment by Xcedf** [comment]
Monkey face is back after #17720, revertig TextureUtils from that fixes it back
regression:
![Image](https://github.com/user-attachments/assets/2954a9f3-74a7-4647-944f-c6c500d9bdfc)
correct image:
![Image](https://github.com/user-attachments/assets/a68d09a8-7465-4a19-8183-5efcbacdb8dd)
Michael's jacket, regression:
![Image](https://github.com/user-attachments/assets/ff4e929c-e208-4c35-8fbb-84fe3aa1476a)
correct image:
![Image](https://github.com/user-attachments/assets/6153096d-1ae2-46e9-b1b6-253dfcc6e916)

**Comment by matiashuarta** [comment]
issues persist as of today, water broken and monkey face issue.

**Comment by digant73** [comment]
please an update on current build

**Comment by MarioSonic2987** [comment]
Reflections on water are still broken:
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/bdccea6e-b4c2-47b9-96f8-ed63ee7efe91" />
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/c62f5716-3ce9-4313-b4ea-171aa4ab5ceb" />

Black texture regression from #17720 is still happening:
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/aac986dd-867a-4377-a18f-902159c003d8" />

[Log](https://github.com/user-attachments/files/29242145/RPCS3.log.gz)

**Comment by digant73** [comment]
@MarioSonic2987 do you confirm that the reflection issue (the last remaining issue in your ticket) is not present on OpenGL? meanwhile I'm waiting for a disc from a colleague, so can you share a game save where the issue is present?

**Comment by MarioSonic2987** [comment]
@digant73 It also happens in OpenGL, but performance tanks A LOT:
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/ce230c9d-689c-4408-a833-a278d4a821bf" />

[Savedata](https://github.com/user-attachments/files/29246211/savedata.GTA.V.23-6-26.zip) - It's the third or fourth save.

You can change the weather with a [cheat code](https://gta.wiki/w/Cheats_in_GTA_V). You need to trigger this many times until you get rainy or stormy weather.

**Comment by digant73** [comment]
ok. you reported reflections are broken only on vulkan. Maybe that was true in the past but now also opengl is broken? if so it could be good to identify the regressing PR

**Comment by MarioSonic2987** [comment]
Enabling all buffer options (WCB+WDB+RCB+RDB) makes the flickering rainbow colors almost disappear. Water is still broken.
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/e0a78210-177e-4b30-b59e-a53fa475c445" />

When you enable **Allow Host GPU labels**, the water breaks even more.
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/9392bb8d-891c-4f03-9b05-6536a4c3709f" />

Black texture regression from https://github.com/RPCS3/rpcs3/pull/17720 is still happening.
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/420fa2e2-df96-45b4-b621-52010996a1ca" />

**Comment by AresOfVinland** [comment]
The the prologue it seems Trevor and Michael get red textures. Attached are photos 👍
This is on Vulkan btw, haven't tested on GL.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/5b46ff95-12ca-4f8a-93b7-ca738456dfb0" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/65269a4a-47a2-4642-a881-028459a23de9" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/2b525658-51ab-48e3-a520-f0f611b58c5b" />

**Comment by MarioSonic2987** [comment]
I can't replicate those red textures.
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/0d7d5d67-70cb-4885-9fb0-066afe9d183f" />
<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/304a0de4-77d7-488f-ae10-5a1eef996764" />

**Comment by joselitovecino-cmyk** [comment]
MarioSonic, could you please fix the car intermitent vertex explosion, there are some configs or ways to fix the water with a config in rpcs3, i hate it, also i know that it happens with write color buffers on, and if i disable it there is no problem, but the water and lighting gets bugged, so could you fix it. Thank you
